### PR TITLE
Add Deep/Shallow Copy methods

### PR DESCRIPTION
**Pull request by kMutagene**
This PR
- Adds distinct methods for:
  - `ShallowCopyDynamicProperties(To)`: The 'old' copy method, does not atttempt to do any copy logic on nested dynamic objects, lists, etc. Also, copied properties are reference equal where applicable (meaning mutating them on the source object will mutate them on the copied object). It is now properly named and has better xml docs
  - `DeepCopyDynamicProperties(To)`: Attempts to 
    - recursively copy dynmic properties
    - create new instances of types  where possible, e.g. for DynamicObj (collections) or classes that implement `ICloneable`
    - Has some implications on derived types from DynamicObj, e.g. them loosing their superclass by being cast to DynamicObj.

### CODE REVIEW

**Comment by kMutagene** [comment]
hopefully only tests left to add

**Comment by kMutagene** [comment]
Decided to leave further changes to `combine` up for a new PR to move with other things, combine does a deep copy first, which should already mitigate some mutation issues.

File: tests/DynamicObject.Tests/Main.fs [hunk 1]
```diff
@@ -4,7 +4,7 @@ open Fable.Pyxpecto
 
 let all = testSequenced <| testList "DynamicObj" [
     ReflectionUtils.Tests.main
-    DynamicObj.Tests.main
+    DynamicObjs.Tests.main
```
Comment (HLWeil) [inline tests/DynamicObject.Tests/Main.fs:7]: Why the "s"?

**Review by HLWeil — CHANGES_REQUESTED** [review]
Looks great! Only two minor requests

File: tests/DynamicObject.Tests/Main.fs (hunk 1 above)
Comment (kMutagene) [reply tests/DynamicObject.Tests/Main.fs:7]: that was the name of the file i refactored into separate ones 🤷

Comment (kMutagene) [reply tests/DynamicObject.Tests/Main.fs:7]: ah also, this: 
>  A namespace and a module named 'DynamicObj.Tests' both occur in two parts of this assembly

File: src/DynamicObj/DynamicObj.fs [hunk 2]
```diff
@@ -224,32 +224,155 @@ type DynamicObj() =
...
     /// </summary>
-    member this.CopyDynamicProperties() =
+    member this.ShallowCopyDynamicProperties() =
         let target = DynamicObj()
-        this.CopyDynamicPropertiesTo(target)
+        this.ShallowCopyDynamicPropertiesTo(target, true)
         target
 
+    // internal helper function to deep copy a boxed object (if possible)
+    static member internal tryDeepCopyObj (o:obj) =
```
[the remark this answers is not in the document]
Comment (kMutagene) [reply src/DynamicObj/DynamicObj.fs:260]: This is so specific, i am not sure if it is _ever_ useful for other purposes. Also, this function uses the DynamicObj class and is used in it

**Pull request merged**